homeassistant/components/inels/entity.py

```diff
@@ -0,0 +1,88 @@
+"""Base class for iNELS components."""
+
+from __future__ import annotations
+
+from inelsmqtt.devices import Device
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, LOGGER
+
+
+class InelsBaseEntity(Entity):
+    """Base Inels device."""
+
+    def __init__(
+        self,
+        device: Device,
+        key: str,
+        index: int,
+    ) -> None:
+        """Init base entity."""
+        self._device: Device = device
+        self._device_id = self._device.unique_id
+        self._attr_name = self._device.title
+
+        self._parent_id = self._device.parent_id
+        self._attr_unique_id = self._device_id  # f"{self._parent_id}-{self._device_id}"
+
+        self._key = key
+        self._index = index
+
+        self._device.add_ha_callback(self.key, self.index, self._callback)
+
+    async def async_added_to_hass(self) -> None:
+        """Add subscription of the data listener."""
+        self._device.mqtt.subscribe_listener(
+            self._device.state_topic, self._device.unique_id, self._device.callback
+        )
+
+        self.async_on_remove(lambda: LOGGER.info("Entity %s to be removed", self.name))
```

Comment:
Why do we log this on remove?

Reply:
I agree that in a production environment, it is unnecessary. I will remove it.